Meadowlark-js/packages/meadowlark-utilities/src/Logger.ts

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import winston from 'winston';
+
+const timestampFormat: string = 'YYYY-MM-DD HH:mm:SS';
+
+const convertErrorToString = (err) => {
+  // Preserve any dictionary, but otherwise convert to string
+  if (err != null && err.constructor !== Object) {
+    return err.toString();
+  }
+  return err;
+};
+
+const format = winston.format.combine(
+  winston.format.label({
+    label: 'Meadowlark',
+  }),
+  winston.format.timestamp({
+    format: timestampFormat,
+  }),
+  winston.format.json(),
+);
+
+const offlineFormat = winston.format.combine(
+  winston.format.timestamp({
+    format: timestampFormat,
+  }),
+  winston.format.printf(({ level, message, timestamp, extra, error }) => {
+    let m = message;
+    let e = error ?? extra ?? '';
+
+    if (typeof message === 'object') {
+      // TypeScript thinks that this is a string, but it there are cases where it ends up being an object
+      m = convertErrorToString(m);
+    }
+    if (typeof e === 'object') {
+      e = JSON.stringify(e);
+    }
+
+    return `${timestamp} ${level} ${m} ${e}`;
+  }),
+  winston.format.colorize({
+    all: true,
+  }),
+);
+
+// Logger begins life "uninitialized" and in silent mode
+let isInitialized = false;
+
+// Create and set up a silent default logger transport - in case a library is using the default logger
+const transport = new winston.transports.Console();
+transport.silent = true;
+winston.configure({ transports: [transport] });
+
+// Set initial logger to silent
+let logger: winston.Logger = winston.createLogger({
+  transports: [transport],
+});
+
+/**
+ * This should be called by frontend services at startup, before logging. Because services can have
+ * multiple startup points (e.g. multiple lambdas), this checks if initialization has already happened.
+ */
+export function initializeLogging(): void {
+  if (isInitialized) return;
+
+  const offline = process.env.IS_LOCAL === 'true';
```

Comment:
I'll create a new ticket to consider moving this to the config abstraction module. RND-435.